planner/core/exhaust_physical_plans.go

```diff
@@ -1969,61 +1969,75 @@ func filterIndexJoinBySessionVars(sc sessionctx.Context, indexJoins []PhysicalPl
 	return indexJoins
 }
 
+func (p *LogicalJoin) prefer(joinFlags ...uint) bool {
```

Comment:
Suggest renaming it like `preferAny` to make it more clear.

Reply:
updated